fitnessllm_dataplatform/stream/strava/services/api_interface.py

```
@@ -0,0 +1,118 @@
+        """Retrieve strava access token."""
+        payload = {
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "code": code,
+            "grant_type": grant_type,
+        }
+        response = requests.post(StravaURLs.AUTH_URL.value, data=payload)
+        if response.status_code != 200:
+            raise Exception(
+                f"Request to strava failed with status {response.status_code}: {response.text}"
+            )
```

Ruff (0.8.2)

Create your own exception

(TRY002)